### PR TITLE
[qtmozembed] Fix loading for pages loaded with pushState. Fixes JB#56556

### PR DESCRIPTION
**Pull request by mkenttala**

### CODE REVIEW

File: src/qmozview_p.cpp [hunk 1]
```diff
@@ -1106,6 +1106,11 @@ void QMozViewPrivate::OnFirstPaint(int32_t aX, int32_t aY)
 #endif
     mIsPainted = true;
     mViewIface->firstPaint(aX, aY);
```
Comment (rainemak) [inline src/qmozview_p.cpp:1108]: This introduces possibility that domContent is not loaded before we got a painted frame (no sure how would I do that but). I'd rather follow ```painted``` property on the browser side.

**Review by rainemak — CHANGES_REQUESTED** [review]
I'd rather fix following lines from browser side:

https://github.com/sailfishos/sailfish-browser/blob/next/apps/core/declarativewebcontainer.cpp#L1041

https://github.com/sailfishos/sailfish-browser/blob/next/apps/core/declarativewebcontainer.cpp#L199

I vaguely remember that as some point of time there was a reason to wait for domContentLoaded and I think it was due to tab creation. As we now create new tabs when requested, without resolving url beforehand, changing above two lines to listen/call firstPaint should be enough and would not change domContentLoaded behaviour.

File: src/qmozview_p.cpp (hunk 1 above)
Comment (rainemak) [reply src/qmozview_p.cpp:1108]: This was it https://github.com/sailfishos/sailfish-browser/pull/485 ( https://github.com/sailfishos/sailfish-browser/commit/0af8ad3563e8b70577de4bb2828f10537f7136ab ).

The issue was other way around that first paint is not always triggered and domContentLoaded is there as a final fallback. 

So adding connection to the firstPaint and extending guards (similarly also within side of the updateActiveTabRendered) should do the same.

```
connect(m_webPage.data(), &QOpenGLWebPage::firstPaint,
              this, &DeclarativeWebContainer::updateActiveTabRendered, Qt::UniqueConnection);
...
if (m_webPage->completed() && m_webPage->active() && (m_webPage->isPainted() || m_webPage->domContentLoaded())) {
    m_webPage->update();
}
```

Comment (rainemak) [reply src/qmozview_p.cpp:1108]: +
```
-    readyToPaint: resourceController.videoActive ? webView.visible && !resourceController.displayOff : webView.visible && webView.contentItem && webView.contentItem.domContentLoaded
+    readyToPaint: resourceController.videoActive ? webView.visible && !resourceController.displayOff : webView.visible && webView.contentItem
+                                                   && (webView.contentItem.domContentLoaded || webView.contentItem.painted)
```

**Comment by mkenttala** [comment]
Moving fix to sailfish-browser.

**Pull request closed without merging**